samza-core/src/main/java/org/apache/samza/table/BaseReadWriteTable.java

```diff
@@ -66,4 +67,12 @@ public void init(Context context) {
   public String getTableId() {
     return tableId;
   }
+
+  public interface Func0 {
+    void apply();
+  }
+
+  public interface Func1<T> {
+    CompletableFuture<T> apply();
```

Comment:
Can't you use a `Supplier<CompletableFuture<T>>` if all you need is an interface that returns a future?